gensim/models/word2vec.py

```diff
@@ -1818,19 +1819,28 @@ def predict_output_word(self, context_words_list, topn=10):
 
         if not hasattr(self.wv, 'vectors') or not hasattr(self, 'syn1neg'):
             raise RuntimeError("Parameters required for predicting the output words not found.")
+        
+        # Retrieve indices if words were passed as input, otherwise keep the input indices
+        # Remark : out-of-vocabulary words are discarded.
+        word2_indices = []
+        max_index = self.wv.vectors.shape[0]
+        for w in context_words_list:
+            if w in self.wv:
```

Comment:
@gojomo I recall doc2vec doing some unintuitive stuff around integer indices … is this construct safe?

Reply:
I don't see a problem (but also per my sibling comment don't see any point to replacing the older 1-liner).